src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatCopy = [...fromFormat];
+  const toFormatCopy = [...toFormat];
+
+  const separator = fromFormatCopy.pop();
+  const parts = date.split(separator);
+
+  const dateParts = {};
+
+  fromFormatCopy.forEach((part, index) => {
+    dateParts[part] = parts[index];
+  });
+
+  if (toFormatCopy.includes('YY') && dateParts['YYYY']) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  }
+
+  if (toFormatCopy.includes('YYYY') && dateParts['YY']) {
+    dateParts['YYYY'] =
+      dateParts['YY'] < '30' ? '20' + dateParts['YY'] : '19' + dateParts['YY'];
+  }
```

Comment:
The conversion logic from 'YY' to 'YYYY' uses a hardcoded threshold ('30'), which might not be suitable for all scenarios. Consider making this logic more flexible or configurable.